### PR TITLE
[Fix](app error handling)

### PR DESCRIPTION
**Pull request by cb-amutha**
Error handling based on response code from Chargebee API and storekit/google billing error codes and also updated appropriate error messages

### CODE REVIEW

**Comment by ciriousjoker** [comment]
It's very useful to access the native store's error codes as well. Could you please pass those as well?

**Example:**
- Play Store returns "BILLING_UNAVAILABLE" (see this issue for more info: https://github.com/chargebee/chargebee-android/issues/63)
- In this case, we want to tell the user to retry the action

File: example/lib/main.dart [hunk 1]
```diff
@@ -302,7 +302,7 @@ class _MyHomePageState extends State<MyHomePage> {
       }
 
     } on PlatformException catch (e) {
-      print('${e.message}, ${e.details}');
+      print('Error Message: ${e.message}, Error Details: ${e.details}, Error Code: ${e.code}');
       if (mProgressBarUtil.isProgressBarShowing()) {
         mProgressBarUtil.hideProgressDialog();
```
Comment (cb-haripriyan) [inline example/lib/main.dart:307]: Would we be able to show how we would want the errors to be handled by using the `httpStatusCode`?

File: ios/Classes/FlutterErrorHandler.swift [hunk 2]
```diff
@@ -10,57 +10,68 @@ import Foundation
...
+        case .invalidCatalogVersion:return 19
+        case .invalidPurchase:return 20
         }
-        
-        return FlutterError(code: errorCode ?? "",
-                            message: message,
-                            details: details)
     }
 }
-
```
Comment (cb-haripriyan) [inline ios/Classes/FlutterErrorHandler.swift:66]: How would this change affect the current error handling logic of the users?

File: android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt [hunk 3]
```diff
@@ -290,7 +290,15 @@ class ChargebeeFlutterSdkPlugin : FlutterPlugin, MethodCallHandler, ActivityAwar
...
     private fun onError(error: CBException, result: Result) {
-        result.error("${error.apiErrorCode}", "${error.message}", error)
+        try {
+            result.error("${error.httpStatusCode}", "${Gson().fromJson(
+                error.message,
+                ErrorDetail::class.java
+            ).message}", error.localizedMessage)
+        }catch (exp: Exception){
+            result.error("${error.httpStatusCode}", "${error.message}", error.localizedMessage)
+        }
```
[the remark this answers is not in the document]
Comment (cb-amutha) [reply android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt:300]: Updated `apiErrorCode` to `httpStatusCode` and still the PlatformException remains same params and current error handling logic should not gets affect as part of this change.

[the remark this answers is not in the document]
Comment (cb-haripriyan) [reply android/src/main/kotlin/com/chargebee/flutter/sdk/ChargebeeFlutterSdkPlugin.kt:300]: Ok. Can we make this explicit in our code comments/readme?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,8 @@
+## 0.0.10
+Bug fixes
+* Fixes product lacking information (#45)
```
Comment (cb-haripriyan) [inline CHANGELOG.md:3]: Adds additional product imformation

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
+## 0.0.10
+Bug fixes
+* Fixes product lacking information (#45) 
+* Reply already submitted and subscriptionId and planId made as non-optional (#48)
```
Comment (cb-haripriyan) [inline CHANGELOG.md:4]: Should this be `replay`? Let's rephrase this.

**Review by cb-haripriyan — APPROVED** [review]

**Pull request merged**